src/backend/cpu/kernel/sift_nonfree.hpp

```diff
@@ -177,12 +177,12 @@ void gaussian1D(T* out, const int dim, double sigma=0.0)
 }
 
 template<typename T>
-Array<T> gauss_filter(float sigma)
+Param<T> gauss_filter(float sigma)
 {
     // Using 6-sigma rule
     unsigned gauss_len = std::min((unsigned)round(sigma * 6 + 1) | 1, 31u);
 
-    Array<T> filter = createEmptyArray<T>(gauss_len);
+    Param<T> filter = createEmptyParam<T>(gauss_len);
```

Comment:
Params should never be created without associated Array<T> objects. They are only used to represent a full array in a more compact structure. They do not manage memory and this is very unsafe. We shouldn't even have the createEmptyParam functions.